### ISSUE TITLE
SC2140: Incorrectly warns on terms between matched pairs of quotes

### ISSUE THREAD

**Issue by srcshelton** [koalaman/shellcheck#2265]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2140
- My shellcheck version (`shellcheck --version` or "online"): Docker container v0.7.2
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /usr/bin/env bash

…

                podman container commit \
                    "${container_name}-${compiler}" \
                    "${container_name%-${package_version}}.${compiler}:${package_version}" \
                    ${IMAGE_ROOT:+--storage-opt="" --root "${IMAGE_ROOT}"} 2>&1 |
                tee -a "log/docker.${container_name}.log"
```

#### Here's what shellcheck currently says:

```
In /mnt/build-kernel.docker line 342:
                                        ${IMAGE_ROOT:+--storage-opt="" --root "${IMAGE_ROOT}"} 2>&1 |
                                                                      ^------^ SC2140: Word is of the form "A"B"C" (B indicated). Did you mean "ABC" or "A\"B\"C"?

For more information:
  https://www.shellcheck.net/wiki/SC2140 -- Word is of the form "A"B"C" (B in...
```

#### Here's what I wanted or expected to see:

No warning - the (empty) quotes associated with passing an empty string to `storage-opt` and the quoted `IMAGE_ROOT` argument for `--root` are both balanced: ` --root ` is not an instance of SC2140.

N.B. This can be trivially fixed in this case by changing the empty pair of quotes to single-quotes.

**Comment by ELLIOTTCABLE** [comment]
I'm also running into this, in almost exactly the same situation (i.e. inside a conditional-expansion of the form `VAR:+...`.)

```sh
mochaify \
   "$unit_dir"/*.tests.coffee "$unit_dir"/*.tests.ts \
   ${integration:+"$integration_dir/"*.tests.coffee "$integration_dir/"*.tests.ts} "$@"
```

```
In /Users/ec/Sync/Code/Paws/Paws.js/Scripts/test.sh line 285:
   ${integration:+"$integration_dir/"*.tests.coffee "$integration_dir/"*.tests.ts} "$@"
                                     ^-------------^ SC2140: Word is of the form "A"B"C" (B indicated).
 Did you mean "ABC" or "A\"B\"C"?
```

In fact, this exact pattern was suggested *elsewhere* by the shellcheck wiki; I can't remember which rule it was, but it's specifically suggested to use `${this:+instead "because"}` it allows proper quote-expansion like the above! (=

**Comment by koalaman** [comment]
You're right, this was the same root cause as #1565 where it incorrectly triggered on regex because the spaces were implicitly included as part of the word. This has now been fixed. Thanks!
